Utils/trigger_private_build.py

```diff
@@ -0,0 +1,125 @@
+
+
+def is_infrastructure_change(modified_files):
+    for file in modified_files:
+        if file in INFRASTRUCTURE_FILES:
+            return True
+
+        path = os.path.dirname(file)
+        for dir_path in INFRASTRUCTURE_FOLDERS:
+            if path.startswith(dir_path):
+                return True
+    return False
+
+
+def get_dispatch_workflows_ids(bearer_token, branch):
+    res = requests.request("GET",
+                           GET_DISPATCH_WORKFLOWS_URL,
+                           headers={'Authorization': bearer_token},
+                           params={'branch': branch, 'event': 'repository_dispatch'},
+                           verify=False)
+    if res.status_code != 200:
+        logging.error(f'Failed to gets private repo workflows, request to '
+                      f'{GET_DISPATCH_WORKFLOWS_URL} failed with error: {str(res.content)}')
+        sys.exit(1)
+
+    try:
+        workflows = json.loads(res.content)
+    except ValueError:
+        logging.error('Enable to parse private repo workflows response')
+        sys.exit(1)
+
+    workflows = workflows.get('workflow_runs', [])
+    return [workflow.get('id') for workflow in workflows]
+
+
+def main():
+    # get github_token parameter
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('--github-token', help='Github token')
+    args = arg_parser.parse_args()
+    bearer_token = 'Bearer ' + args.github_token
+
+    # get branch name
+    branches = tools.run_command("git branch")
+    branch_name_reg = re.search(r"\* (.*)", branches)
+    branch_name = branch_name_reg.group(1)
+
+    files = get_modified_files(branch_name)
+
+    if is_infrastructure_change(files):
+        # get the workflows ids before triggering tye build
+        workflow_ids = get_dispatch_workflows_ids(bearer_token, 'master')
+
+        # trigger private build
+        payload = {'event_type': f'Trigger private build from content/{branch_name}',
+                   'client_payload': {'branch': branch_name, 'is_infra_build': 'True'}}
+
+        res = requests.request("POST",
+                               TRIGGER_BUILD_URL,
+                               headers={'Accept': 'application/vnd.github.everest-preview+json',
+                                        'Authorization': bearer_token},
+                               data=json.dumps(payload),
+                               verify=False)
+
+        if res.status_code != 204:
+            logging.error(f'Failed to trigger private repo build, request to '
+                          f'{TRIGGER_BUILD_URL} failed with error: {str(res.content)}')
+            sys.exit(1)
+
+        # wait 5 seconds and get the workflow ids again
+        time.sleep(5)
+        workflow_ids_new = get_dispatch_workflows_ids(bearer_token, 'master')
+
+        # compare with the first workflows list to get the current id
+        workflow_id = [x for x in workflow_ids_new if x not in workflow_ids]
+        if workflow_id:
```

Comment:
what if you don't find the new workflow?

Reply:
Or what if more than one workflow was triggered at the same time? Is there a way to get the specific workflow? like adding something unique in the post request body and then find it in the get request?

Reply:
@hod-alpert [create repo dispatch event](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event) returns nothing, this is why we need to call [list workflows](https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository) command twice to gets the workflow id